Switch Jenkins image tracking to registry tag due to Samples Operator… … deprecation (#1151)
coreos/fedora-coreos-pipeline: HACKING.md

    jenkins.io/credentials-description="GitHub coreosbot token as username/password"
```

### Create root CA certificate secret

The root CA certificate (ca.crt) is required and should be created as a secret.
This example assumes that the ca.crt file is present in your current working
directory.

> If you are working in a development environment, you still need to create a
> dummy configuration file as shown below:

Reviewer: ```suggestion
If you are working in an environment that doesn't need a custom root CA, you still need to create a dummy configuration file as shown below:
```

3. the Jenkins agent BuildConfig (if a root CA cert was provided),
4. the jenkins-config configmap.

> Note: If you are working with the Fedora staging pipeline, use the staging branch to define the pipecfg.

Reviewer: ```suggestion
Note: If you are working with the Fedora staging pipeline, specify the `staging` branch for the pipecfg: `--pipecfg https://github.com/coreos/fedora-coreos-pipeline@staging`.
```

oc start-build --follow jenkins-agent-base-with-cert
```

If you're working with the Staging environment, keep in mind that the registry

> is not well maintained. This can lead to issues due to mismatched agent images.

Author: Yes, that was my understanding, but that does not seems to be case for openshift namespace in the stg.  We need to do a follow up for it for sure.  However, the tags are not tracking the changes right, leading to the  issue we are seeing, the agent image still using Java 11, when the latest one are in Java 17 and 21. There are some tag with the buildidcommit from years ago.

> is not well maintained. This can lead to issues due to mismatched agent images.

Author: As you can see `scheduled-upgrade`" https://github.com/openshift/cluster-samples-operator/blob/main/assets/operator/ocp-x86_64/jenkins/imagestreams/jenkins-agent-base-rhel.json#L52-L62 is tracking the same tag I added.  The real issue is, latest is not being updated:

https://github.com/openshift/cluster-samples-operator/blob/main/assets/operator/ocp-x86_64/jenkins/imagestreams/jenkins-agent-base-rhel.json#L14-L24

That's probably a matter for changing it to point to the other one in stg
